chain/beacon/store.go

```diff
@@ -43,12 +45,44 @@ func (a *appendStore) Put(b *chain.Beacon) error {
 	if b.Round != a.last.Round+1 {
 		return fmt.Errorf("invalid round inserted: last %d, new %d", a.last.Round, b.Round)
 	}
-	if !bytes.Equal(a.last.Signature, b.PreviousSig) {
+	if err := a.Store.Put(b); err != nil {
+		return err
+	}
+	a.last = b
+	return nil
+}
+
+// schemeStore is a store that run different checks depending on what scheme is being used.
+type schemeStore struct {
+	chain.Store
+	sch  scheme.Scheme
+	last *chain.Beacon
+	sync.Mutex
+}
+
+func newSchemeStore(s chain.Store, sch scheme.Scheme) chain.Store {
+	last, _ := s.Last()
+	return &schemeStore{
+		Store: s,
+		last:  last,
+		sch:   sch,
+	}
+}
+
+func (a *schemeStore) Put(b *chain.Beacon) error {
+	a.Lock()
+	defer a.Unlock()
+
+	if a.sch.DecouplePrevSig {
+		b.PreviousSig = nil
```

Comment:
Let's clearly writes off  in a comment that this store sets this field to nil in case we are in unchained mode,

Reply:
Done!